OutOfSchool/OutOfSchool.BusinessLogic/Services/CompetitiveEventDrafts/CompetitiveEventDraftService.cs

```diff
@@ -0,0 +1,733 @@
+        }
+
+        await competitiveEventDraftRepository.Delete(competitiveEventDraft);
+
+        logger.LogDebug("Draft was successfully approved and deleted. Draft Id = {DraftId}.", id);
+    }
+
+    // <inheritdoc/>
+    public async Task Reject(Guid id, string rejectionMessage)
+    {
+        logger.LogDebug("Rejecting CompetitiveEventDraft started. CompetitiveEventDraft Id = {id}.", id);
+
+        var competitiveEventDraft = await GetDraftById(id);
+
+        if (competitiveEventDraft.DraftStatus != CompetitiveEventDraftStatus.PendingModeration && competitiveEventDraft.DraftStatus != CompetitiveEventDraftStatus.EditedByModerator)
+        {
+            throw new ArgumentException("This CompetitiveEventDraft can`t be rejected.");
+        }
+
+        competitiveEventDraft.DraftStatus = CompetitiveEventDraftStatus.Rejected;
+        competitiveEventDraft.RejectionMessage = rejectionMessage;
+
+        await competitiveEventDraftRepository.Update(competitiveEventDraft);
+        logger.LogDebug("CompetitiveEventDraft was successfully rejected. Draft Id = {id}.", id);
+    }
+
+    // <inheritdoc/>
+    public async Task<CompetitiveEventV2Dto> UpdateCompetitiveEvent(CompetitiveEventV2Dto competitiveEventV2Dto)
+    {
+        logger.LogDebug("Competitive event updating process started. CompetitiveEvent Id = {Id}.", competitiveEventV2Dto.Id);
+
+        var existingCompetitiveEvent = await competitiveEventService.GetById(competitiveEventV2Dto.Id);
+
+        if (existingCompetitiveEvent == null)
+        {
+            throw new InvalidOperationException($"There is no CompetitiveEvent with such Id. CompetitiveEvent can`t be updated.");
+        }
+
+        var draft = await competitiveEventDraftRepository.Get(whereExpression: ced => ced.Id == competitiveEventV2Dto.Id)
+            .AsNoTracking()
+            .FirstOrDefaultAsync();
+
+        if (draft != null)
+        {
+            logger.LogDebug("CompetitiveEvent draft for this CompetitiveEvent exists. CompetitiveEvent can`t be updated. CompetitiveEvent Id = {Id}.", competitiveEventV2Dto.Id);
+
+            throw new InvalidOperationException("CompetitiveEvent draft for this CompetitiveEvent exists. CompetitiveEvent can`t be updated.");
+        }
+
+        if (AreModeratedFieldsChanged(competitiveEventV2Dto, existingCompetitiveEvent))
+        {
+            logger.LogDebug("Moderated fields was changed. CompetitiveEvent draft creation initiated. CompetitiveEvent Id = {Id}.", competitiveEventV2Dto.Id);
+            return (await Create(competitiveEventV2Dto)).CompetitiveEventDraft.CompetitiveEventDetails;
+        }
+
+        logger.LogDebug("Moderated fields was not changed. CompetitiveEvent update initiated. CompetitiveEvent Id = {Id}.", competitiveEventV2Dto.Id);
+
+        return competitiveEventService.UpdateV2(competitiveEventV2Dto.ToDraft().ToV2CreateRequestDto()).Result.CompetitiveEventV2;
```

Comment:
also agree with code rabbit, that it is better to use await here. might add an extra line of code, but would be safer